astropy/table/column.py

```diff
@@ -422,7 +422,7 @@ def format(self, format_string):
         try:
             # test whether it formats without error exemplarily
             self.pformat(max_lines=1)
-        except TypeError as err:
+        except (TypeError, ValueError) as err:
```

Comment:
Shouldn't we just catch any exception here? I.e., `except Exception as err`?

Also, I may have worried about this before, but what happens if the first element is masked? I vaguely recall writing delayed formatting for that case, and `max_lines=1` might then not catch it. Perhaps too much of a detail, though.

Reply:
So, I should read all my e-mail before commenting. So indeed I wrote about this 24 days ago (just tells you how long my memory lasts these days...)

Reply:
Bottom line: happy with this, though I'd use `Exception` - seems justified to be general in this case, since we re-raise it as a `ValueError` anyway.